src/redux/stores/balances/balancesSubscriber.ts

```diff
@@ -15,6 +20,17 @@ export const configureBalancesSubscriber = () => {
     const lastTransaction = transactions.transactions[0];
     const lastSucceededTransaction = lastTransaction?.status === 'succeeded' ? lastTransaction : undefined;
 
+    if (web3.chainId && chainId !== web3.chainId && web3.chainId !== config.chainId) {
```

Comment:
Why is `chainId` stored in balance store? If it is a duplicate of `chainId` from web3 store should we remove one of them?

Reply:
It's not stored in balances store. Subscriber only listens to changes in store, and the chainId is only stored in the function.